api/src/main/java/org/openmrs/module/commonreports/reports/AntenatalRisksReportManager.java

```diff
@@ -0,0 +1,254 @@
+		CompositionCohortDefinition ccd = new CompositionCohortDefinition();
+		ccd.initializeFromElements(_prenatal, anemia, female);
+		antenatalRisks.addRow(MessageUtil.translate("commonreports.report.antenatalRisks.ironDefANC"), ccd,
+		    parameterMappings);
+		
+		// Prenatal visit + Fer Folate Co prescribed 
+		SqlCohortDefinition sqd = new SqlCohortDefinition(
+		        "select patient_id from orders where concept_id=(select DISTINCT concept_id from concept_name where name = 'Ferrous folate' and locale='en' and concept_name_type='FULLY_SPECIFIED' and voided = 0) and order_type_id =(select order_type_id from order_type where name = 'Drug Order')");
+		sqd.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		sqd.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		CompositionCohortDefinition ccd1 = new CompositionCohortDefinition();
+		ccd1.initializeFromElements(_prenatal, sqd, female);
+		antenatalRisks.addRow(MessageUtil.translate("commonreports.report.antenatalRisks.prenatalIron"), ccd1,
+		    parameterMappings);
+		
+		//Prenatal visit treated for Fe def (Same as above--> Prenatal visit + Fer Folate Co prescribed)
+		antenatalRisks.addRow(MessageUtil.translate("commonreports.report.antenatalRisks.prenatalIronDef"), ccd1,
+		    parameterMappings);
+		
+		//Mothers with a birth plan
+		SqlCohortDefinition scd = new SqlCohortDefinition();
+		String st = "select person_id from obs where concept_id = (select DISTINCT concept_id from concept_name where name = 'Number of Weeks' and locale='en' and concept_name_type='FULLY_SPECIFIED' and voided = 0)  AND obs_group_id IN (select obs_id from obs where concept_id = (select DISTINCT concept_id from concept_name where name = 'Estimated Gestational Age' and locale='en' and concept_name_type='FULLY_SPECIFIED' and voided = 0))";
+		scd.setQuery(st);
+		scd.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		scd.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		antenatalRisks.addRow(MessageUtil.translate("commonreports.report.antenatalRisks.motherBirthPlan"), scd,
+		    parameterMappings);
+		
+		//Prenatal visit + malaria test positive + chloroqine co prescribed 
+		CodedObsCohortDefinition malaria = new CodedObsCohortDefinition();
+		malaria.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		malaria.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		malaria.setOperator(SetComparator.IN);
+		malaria.setQuestion(cs.getConcept("Malaria"));
+		
+		List<Concept> malariaPositiveConceptSet = new ArrayList<Concept>();
+		malariaPositiveConceptSet.add(cs.getConcept("Positif"));
+		malariaPositiveConceptSet.add(cs.getConcept("Un plus"));
+		malariaPositiveConceptSet.add(cs.getConcept("Deux plus"));
+		malariaPositiveConceptSet.add(cs.getConcept("Trois plus"));
+		malariaPositiveConceptSet.add(cs.getConcept("Quatre plus"));
```

Comment:
Could you provide these concepts through configuration?

Reply:
Done